### DISCUSSION TITLE
Frequency of updates of Energy_Produced and Electricity_Consumed values

### DISCUSSION THREAD

**Discussion by buennerbernd** [Arnold-n/P1P2MQTT#140] [Q&A]
Hello,

I'm calculating the current energy production and consumption based on the values Energy_Produced_Heatpump_Total and Electricity_Consumed_Total.
Usually I get updates on these values only every 2 hours.
Last week I was surprised that I suddenly get the values in a higher frequency:
- Energy_Produced_Heatpump_Total: 4 to 6 times per hour
- Electricity_Consumed_Total: every hour

My first assumption was that there was an software update on the heatpump, probably not.
Yesterday I did some minor changes at heatpumps interface. The heatpump made a restart and the higher frequency was gone.
Again I get updates on these values only every 2 hours.

I found out that the good times started with a defrost event.
![image](https://github.com/user-attachments/assets/645b473f-349a-4aa2-a491-6cc585bba0cd)
Red line: Energy_Produced
Green line: Electricity_Consumed

My system: Altherma 3 H HT with Madoka as room thermostat in the main room, only one main zone.

**Has anyone an idea how I can get back the higher update frequency?
What are your observations?**

**Comment by Arnold-n** [comment]
Strange and interesting. What I have seen is that
-on older systems, both energy_produced and electricity_consumed have poor resolution: 1kWh, and energy_produced updates whenever a new kWh of energy is produced, but electricity_consumed is updated exactly once/hour regardless of when the counter reaches a new value, making resolution effectively even worse,
-on newer systems, the update frequency for electricity_consumed is once every 2 hours (while I think energy_produced is still updated in real-time).

So for some strange reason your system switched (last week, suddenly, no changes from your side?) from the newer (worse) mode to the older reporting mode?

As a side note, the bridge (at least more recent versions - you are running an older one) reports Production_Heatpump which is a real-time calculation based on LWT, RWT, and flow which is much more accurate than using energy_produced.

**Reply by buennerbernd** [reply]
Last week I changed a lot, I have enabled the Madoka as room thermostat. Before it was just a dump temperature sensor at the wall.
For that the heatpump restarted several times and I used different users 5678 and 1234.
On the day I observed the effect described above, I did not any changes. After 8 am nobody was at home. Really strange.

I think I will update the bridge in the near future. I just waited for the smart grid option. ;)
